stan/math/opencl/prim/cols.hpp

```diff
@@ -0,0 +1,27 @@
+#ifndef STAN_MATH_OPENCL_PRIM_COLS_HPP
+#define STAN_MATH_OPENCL_PRIM_COLS_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/matrix_cl.hpp>
+#include <stan/math/opencl/rev/matrix_cl.hpp>
+
+namespace stan {
+namespace math {
+/** \ingroup opencl
+ * Returns the number of columns in matrix_cl.
+ *
+ * @tparam T type of elements in the input matrix
+ * @param x the input matrix_cl
+ *
+ * @return number of columns in x
+ */
+template <typename T>
+inline int cols(const matrix_cl<T>& x) {
```

Comment:
Should this also be for matrix_cl expressions?

Reply:
Ah yes, of course.